src/openms/source/QC/TIC.cpp

```diff
@@ -46,9 +46,36 @@ namespace OpenMS
   {
     results_.clear();
   }
-  void TIC::compute(const MSExperiment &exp, float bin_size)
+
+  TIC::Result TIC::compute(const MSExperiment& exp, float bin_size)
   {
     results_.push_back(exp.getTIC(bin_size));
+    struct Result result;
+    const MSChromatogram& tic = exp.getTIC();
+    if (!tic.empty())
+    {
+    for (const auto& p : tic)
+    {
+      result.intensities.push_back(p.getIntensity());
+      result.retention_times.push_back(p.getRT());
+    }
+
+    result.area = result.intensities[0];
+
+    for (UInt i = 1; i < result.intensities.size(); ++i)
```

Comment:
use `size_t` for comparison to `size()`, since UInt != size_t on some platforms (e.g. Windows)